lib/ts_utils/metadata.py

```diff
@@ -300,6 +301,22 @@ def read_metadata(distribution: str) -> StubMetadata:
     )
 
 
+def update_metadata(distribution: str, **new_values: object) -> tomlkit.TOMLDocument:
```

Comment:
I could write an entire `TypedDict` that duplicates the `StubMetadata` for additional type safety, but none of the existing code relied on an exactly typed dict nor validated the input. So technically we're no loosing on anything, and can be considered an improvement for later if we deem it worthwile.